### ISSUE TITLE
How do I sort in Collection.find_one_and_update?

### ISSUE THREAD

**Issue by tyteen4a03** [twisted/txmongo#242]
I have the following code:

            next_job = yield self.scrapers_coll.find_one_and_update(
                {"state": "enqueued"},
                {
                    "$set": {
                        "startTime": time.time(),
                    }
                },
                sort=[("createdOn", pymongo.ASCENDING)]
        )

This gives me a `TypeError: list indices must be integers or slices, not str`.

From going through the source code it looks like TxMongo wants a different format than PyMongo. How do I specify the sort order in this case?

**Comment by IlyaSkriblovsky** [comment]
Yes, txmongo has its own syntax for sort, hint and other query options:

```python
from txmongo import filter

next_job = yield self.scrapers_coll.find_one_and_update(
    {"state": "enqueued"},
    {
        "$set": {
            "startTime": time.time(),
        }
    },
    sort=filter.sort(filter.ASCENDING("createdOn"))
)
```

This way you can sort by multiple fields:
```python
filter.sort(filter.ASCENDING("priority") + filter.ASCENDING("createdOn"))
```

**Comment by tyteen4a03** [comment]
Any plans to update the code to support standard pymongo syntax too?
